### PR TITLE
[6X_STABLE] [backport] CREATE EXTENSION in explicit transaction fails on assertion on executors

### PR DESCRIPTION
**Pull request by QingMa-gp**
Back-port of https://github.com/greenplum-db/gpdb/pull/14534.
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/commands/extension.c [hunk 1]
```diff
@@ -1303,8 +1315,7 @@ CreateExtension(CreateExtensionStmt *stmt)
 	 * create only one extension at the same time.
 	 * Except that QE do CREATE_EXTENSION_BEGIN.
```
Comment (zhrt123) [inline src/backend/commands/extension.c:1316]:
```suggestion
```

File: src/test/modules/test_extensions/Makefile [hunk 2]
```diff
@@ -3,10 +3,12 @@
...
 
-EXTENSION = test_ext_cor test_ext_cine
-DATA = test_ext_cor--1.0.sql test_ext_cine--1.0.sql test_ext_cine--1.0--1.1.sql
+EXTENSION = test_ext_cor test_ext_cine test_ext_cau
+DATA = test_ext_cor--1.0.sql test_ext_cine--1.0.sql test_ext_cine--1.0--1.1.sql \
+       test_ext_cau--1.0--1.1.sql test_ext_cau--1.0.sql \
+       test_ext_cau--1.1.sql test_ext_cau--unpackaged--1.0.sql
 
-REGRESS = test_extensions
+REGRESS = test_extensions #test_extdepend TODO: enable this test after https://github.com/greenplum-db/gpdb/issues/14532
```
[the remark this answers is not in the document]
Comment (QingMa-gp) [reply src/test/modules/test_extensions/Makefile]: Done.

**Comment by kainwen** [comment]
@HustonMmmavr 
seems all the comments have been addressed, would you please take a look at this again? Thanks.

**Review by zhrt123 — APPROVED** [review]

**Review by HustonMmmavr — APPROVED** [review]

**Comment by QingMa-gp** [comment]
Thanks for the review.

**Pull request merged**